documentation/source/quickstart.rst

```diff
@@ -210,11 +210,26 @@ the various actions described in :ref:`assignment-methods` can be used.
     dut.my_signal <= Freeze()     # my_signal stays at current value until released
 
 
+.. _quickstart_reading_values:
+
 Reading values from signals
 ---------------------------
 
-Accessing the :attr:`~cocotb.handle.NonHierarchyObject.value` property of a handle object will return a :any:`BinaryValue` object.
-Any unresolved bits are preserved and can be accessed using the :attr:`~cocotb.binary.BinaryValue.binstr` attribute,
+Values in the DUT can be accessed with the :attr:`~cocotb.handle.NonHierarchyObject.value`
+property of a handle object.
+A common mistake is forgetting the ``.value`` which just gives you a reference to a handle
+(useful for defining an alias name), not the value.
+
+The Python type of a value depends on the handle's HDL type:
+
+| Arrays of ``logic`` and subtypes of that (``sfixed``, ``unsigned``, etc.) are of type :class:`~cocotb.binary.BinaryValue`.
+| Integer nets and constants (``integer``, ``natural``, etc.) return :class:`int`.
+| Floating point nets and constants (``real``) return :class:`float`.
+| Boolean nets and constants (``boolean``) return :class:`bool`.
+| String nets and constants (``string``) return :class:`bytes`.
```

Comment:
Possibly more suited to a list:
```suggestion
* Arrays of ``logic`` and subtypes of that (``sfixed``, ``unsigned``, etc.) are of type :class:`~cocotb.binary.BinaryValue`.
* Integer nets and constants (``integer``, ``natural``, etc.) return :class:`int`.
* Floating point nets and constants (``real``) return :class:`float`.
* Boolean nets and constants (``boolean``) return :class:`bool`.
* String nets and constants (``string``) return :class:`bytes`.
```

Reply:
I agree. [The block](https://cocotb--2376.org.readthedocs.build/en/2376/quickstart.html#reading-values-from-signals) isn't teh prettiest.

Reply:
Agreed, changed.